shared/actions/config.js

```diff
@@ -196,12 +198,97 @@ function _bootstrapSuccess(action: ConfigGen.BootstrapSuccessPayload, state: Typ
   return Saga.all(actions)
 }
 
+function _pgpSecurityModelChangeMessageSaga() {
+  RPCTypes.pgpPgpStorageDismissRpcPromise().catch(err => {
+    console.warn('Error in sending pgpPgpStorageDismissRpc:', err)
+  })
+}
+
+function _loadAvatarHelper(action: {payload: {names: Array<string>, endpoint: string, key: string}}) {
+  const {names, endpoint, key} = action.payload
+  return Saga.all([
+    Saga.call(RPCTypes.apiserverGetRpcPromise, {
+      args: [{key, value: names.join(',')}, {key: 'formats', value: 'square_360,square_200,square_40'}],
+      endpoint,
+    }),
+    Promise.resolve(names),
+  ])
+}
+
+function _afterLoadAvatarHelper([response: {body: string}, names]) {
+  const nameToUrlMap = JSON.parse(response.body).pictures.reduce((map, picMap, idx) => {
+    const name = names[idx]
+    const urlMap = {
+      ...(picMap['square_200'] ? {'200': picMap['square_200']} : null),
+      ...(picMap['square_360'] ? {'360': picMap['square_360']} : null),
+      ...(picMap['square_40'] ? {'40': picMap['square_40']} : null),
+    }
+    map[name] = urlMap
+    return map
+  }, {})
+
+  return Saga.put(ConfigGen.createLoadedAvatars({nameToUrlMap}))
+}
+
+let _avatarsToLoad = {}
+function* _loadAvatars(action: ConfigGen.LoadAvatarsPayload) {
```

Comment:
avatars now use redux instead of their own little weird thing (util/pictures and registering callbacks back and forth and etc)